### ISSUE TITLE
page content erased and changed to "false"

### ISSUE THREAD

**Issue by flallo75** [turnermm/ckgedit#244]
If open a a page in ckedit and try to switch to DW editor the page becomes almost empty, with the only word "false" in the first row. If I want the old page I have to restore a previous revision of the page.

**Comment by turnermm** [comment]
Probably a javascript error.  But you will have to check your browser's javascript console to see what the error  is.  
What version of ckgedit are you using? what version of dokuwiki? and what browser and what operating system is your browser running on?

**Comment by flallo75** [comment]
The page was pasted from a simple (only headings, bulleted list and text color) msword file and after cleaning the code of the page (removing with notepad format markup who define font size and color...) the problme was also removed. If it happens again with some page written directly with ckedit or dw editor I'll let yu know.

by now thanks for your reply

system information:
dokuwiki: Release 2016-06-26a "Elenor of Tsort"
ckgeditor: _ckgDoku-ckedit_457-16-Oct_21-13_03 (but the same problem also on main fork of ckedit)
OS: windows 10 Pro 64bit
browser: firefox 49.0.2
java R.E.: 1.8.0_111 (no errors in console)

**Comment by turnermm** [comment]
The toolbar has a tool for pasting from ms word:
![paste_from_word](https://cloud.githubusercontent.com/assets/176447/19933183/84c62198-a0e0-11e6-8369-a4d9b2fc15dc.png)

It preserves most of the formatting.  However, ms word uses a <font> tag which conflicts with ckgedit's font tag.  So, when you save an ms word page with font and color settings, you might find the font settings in your text, but when you reopen the page in ckgedit, ckgedit will probably remove the font settings from your text.

**Comment by tholu** [comment]
This happens for me as well (latest Dokuwiki, latest Plugin, latest Chrome on OSX). Has nothing to do with formatting from copy+paste (also happens on a correctly formatted wiki page). Will try to track down the root cause.

**Comment by tholu** [comment]
Found the issue: Cloudflare Email Obfuscation lead to a JavaScript Error when parsing the Wikitext:

```
Uncaught Parse Error: <a class="__cf_email__" data-cfemail="2c5f55415c5843414d6c40434745025f55415c5843414d024f4341" href="/cdn-cgi/l/email-protection">[email&nbsp;protected]</a><script data-cfhash='f9e31' type="text/javascript">/* <![CDATA[ */!function(t,e,r,n,c,a,p){try{t=document.currentScript||function(){for(t=document.getElementsByTagName('script'),e=t.length;e--;)if(t[e].getAttribute('data-cfhash'))return t[e]}();if(t&&(c=t.previousSibling)){p=t.parentNode;if(a=c.getAttribute('data-cfemail')){for(e='',r='0x'+a.substr(0,2)|0,n=2;a.length-n;n+=2)e+='%'+('0'+('0x'+a.substr(n,2)^r).toString(16)).slice(-2);p.replaceChild(document.createTextNode(decodeURIComponent(e)),c)}p.removeChild(t)}}catch(u){}}()/* ]]> */</script></li>
HTMLParser.HTMLParser	@	js.php?t=dokuwiki&tseed=8478b59…:11
parse_wikitext @ parse_wiki-cmpr.js:1
(anonymous) @ parse_wiki-cmpr.js:1
dispatch @ js.php?t=dokuwiki&tseed=8478b59…:1
r.handle @ js.php?t=dokuwiki&tseed=8478b59…:1
```

Disabled it for the Wiki by using a Page Rule.

**Comment by turnermm** [comment]
Yes, that problem is (almost) always a result of a javascript error.  Thanks.

**Comment by tholu** [comment]
@turnermm Perhaps it's possible to solve this by removing all <script>-Tags before parsing?

**Comment by simon-roderus** [comment]
I had the same problem with my wiki farm (branch: farms_4.7)

**Comment by turnermm** [comment]
@simon-roderus But the question is: what is the source of the Javascript error.  If you can pinpoint t he error in ckgedit then it can be fixed, but as in the case of @tholu if the error is from something else then that something else has to be fixed.
@tholu I'm not sure what you  have in mind.  Script tags are required when the js is in the body of the html document.

**Comment by tholu** [comment]
@turnermm In my case the problem seems to be, that the automatically inserted <script>-Tag in the source leads to the parsing error. It's not the <script>-Tag itself, that causes this parsing error, but the HTMLParser, which fails to parse the <script>-Tag. 

So if <script>-Tags in the source are a regular occurrence and they should be parseable by HTMLParser, then HTMLParser needs to be fixed to successfully parse these <script>-Tags. If <script>-Tags in general should not be in the source, then a solution would be to remove them before parsing.

**Comment by turnermm** [comment]
I would have to see an example.  ckgedit itself doesn't insert script tags in its source output.  Whatever script tags ckgedit uses are external to the document being parsed.

**Comment by turnermm** [comment]
Closed in favor of #251

**Comment by vvzh** [comment]
This can be caused by reverse proxy setup, see https://github.com/turnermm/ckgedit/issues/335#issuecomment-389122601 for a possible solution.
